### PR TITLE
nixborg: reference implementation for upcoming RFC on hydra PR testing

### PR DESCRIPTION
**Pull request by globin**
###### Motivation for this change

@nixborg consists of three seperate services:
 - nixborg: flask app to receive github hooks and fill a queue (currently redis, probably switching to rabbitmq in the future
- nixborg-workers: workers consuming the queue, doing the git handling, commenting on github, HTTP requesting to nixbot-receiver
- nixbot-receiver: stdlib-dependency-only python script to be run on hydra that calls out to hydra-update-jobset (PR for that cmdline utility upcoming on the hydra repository)

Code at: https://github.com/mayflower/nixborg (to be moved to the NixOS org if implementation is accepted)
PRs to see output: https://github.com/NixOS/nixpkgs/pull/30767, https://github.com/NixOS/nixpkgs/pull/30546
Hydra PR: https://github.com/NixOS/hydra/pull/516
RFC upcoming

cc @edolstra, @grahamc, @fpletz, @7c6f434c, @LnL7, @dezgeg

### CODE REVIEW

**Comment by gebner** [comment]
Is this bot related to @GrahamcOfBorg?

**Review by orivej — APPROVED** [review]

File: nixos/modules/services/web-apps/nixborg.nix [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+      githubToken = mkOption {
+        type = types.str;
+        description = "The bot's github user account token.";
+        example = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
+      };
+
+      githubSecret = mkOption {
+        type = types.str;
+        description = "The github webhook's secret key used for autheticating webhooks";
```
Comment (c0bw3b) [inline nixos/modules/services/web-apps/nixborg.nix]: for _authenticating_ webhooks

**Comment by edolstra** [comment]
It would be better to put this in the nixborg repo and enable it via `fetchGit`.

**Comment by globin** [comment]
Will do

**Pull request closed without merging**